src/content_encoding.rs

```diff
@@ -0,0 +1,38 @@
+use crate::error::{error, no_error, ErrorIterator, ValidationError};
+use serde_json::Value;
+use std::collections::HashMap;
+
+pub(crate) type ContentTypeCheckType = for<'a> fn(&'a Value, &str) -> ErrorIterator<'a>;
+pub(crate) type ContentTypeConverterType =
+    for<'a> fn(&'a Value, &str) -> Result<String, ValidationError<'a>>;
+
+pub fn is_base64<'a>(instance: &'a Value, instance_string: &str) -> ErrorIterator<'a> {
+    if base64::decode(instance_string).is_err() {
+        return error(ValidationError::format(instance, "base64"));
+    }
+    no_error()
+}
+
+pub fn from_base64<'a>(
+    instance: &'a Value,
+    instance_string: &str,
+) -> Result<String, ValidationError<'a>> {
+    match base64::decode(instance_string) {
+        Ok(value) => Ok(String::from_utf8(value)?),
+        Err(_) => Err(ValidationError::format(instance, "base64")),
+    }
+}
+
+lazy_static::lazy_static! {
+    pub(crate) static ref DEFAULT_CONTENT_ENCODING_CHECK: HashMap<&'static str, ContentTypeCheckType> = {
+        let mut map: HashMap<&'static str, ContentTypeCheckType> = HashMap::with_capacity(1);
+        let _ = map.insert("base64", is_base64);
+        map
+    };
+
+    pub(crate) static ref DEFAULT_CONTENT_ENCODING_CONVERT: HashMap<&'static str, ContentTypeConverterType> = {
+        let mut map: HashMap<&'static str, ContentTypeConverterType> = HashMap::with_capacity(1);
+        let _ = map.insert("base64", from_base64);
```

Comment:
To we need `_` variable?

Reply:
I'm usually used to have [`unused-results`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unused-results) lint enable to prevent myself from forgetting to use the result of a method. This means that I do need to assign the result to something and `_` is the usual "trash-bin".

Definitely this boilerplate will be removed into the more final solution (as in this repo the lint is not enforced)